sql-plugin/src/main/scala/com/nvidia/spark/rapids/shuffle/BufferSendState.scala

```diff
@@ -119,6 +128,7 @@ class BufferSendState(
       throw new IllegalStateException("ALREADY CLOSED!")
     }
     isClosed = true
+    cleanMemoryBuffers()
```

Comment:
we should move the `cleanMemoryBuffers` call within `releaseAcquiredToCatalog`. Otherwise, we will hold onto the materialized buffer (and in GDS'es case, the copy to device memory) until `close`, instead of until the UCX callback says its ok to let go.